infrastructure/main.tf

```diff
@@ -49,7 +49,7 @@ resource "azurerm_resource_group" "rg" {
 }
 
 resource "azurerm_storage_account" "provider" {
-  name                      = "${substr(local.account_name, 0, min(length(local.account_name), 24))}"
+  name                      = "${local.account_name}"
```

Comment:
Could you do it like this: https://github.com/hmcts/draft-store/blob/master/infrastructure/main.tf#L29 ?

When in preview, we don't need to append the environment bit, which is most likely to make the name too long, given the product gets the `PR_123` prefix.